src/components/MessageList/MessageList.js

```diff
@@ -121,91 +121,155 @@ const MessageList = ({ messages, handleGoBack }) => {
-                            handleEmojiClick(
-                              reaction,
-                              msg,
-                              !isSameUser(reaction, authenticatedUserUsername)
-                            )
-                          }
-                        >
-                          <Markdown body={reaction.name} />
-                          <p>{reaction.count}</p>
-                        </MessageReactions.Reaction>
-                      ))}
-                  </MessageReactions>
+                  {msg.t === undefined ? (
+                    <>
+                      <Message.Header>
+                        <Message.Name>{msg.u?.name}</Message.Name>
+                        <Message.Username>@{msg.u.username}</Message.Username>
+                        <Message.Timestamp>
+                          {format(new Date(msg.ts), 'h:mm a')}
+                        </Message.Timestamp>
+                        {msg.editedAt && (
+                          <Icon mie="x4" opacity={0.5} name="edit" size="x16" />
+                        )}
+                      </Message.Header>
+                      <Message.Body>
+                        {msg.attachments && msg.attachments.length > 0 ? (
+                          <Attachments attachments={msg.attachments} />
+                        ) : (
+                          <Markdown body={msg.msg} />
+                        )}
+                      </Message.Body>
+                      <MessageReactions>
+                        {msg.reactions &&
+                          serializeReactions(msg.reactions).map((reaction) => (
+                            <MessageReactions.Reaction
+                              key={reaction.name}
+                              mine={isSameUser(
+                                reaction,
+                                authenticatedUserUsername
+                              )}
+                              onClick={() =>
+                                handleEmojiClick(
+                                  reaction,
+                                  msg,
+                                  !isSameUser(
+                                    reaction,
+                                    authenticatedUserUsername
+                                  )
+                                )
+                              }
+                            >
+                              <Markdown body={reaction.name} />
+                              <p>{reaction.count}</p>
+                            </MessageReactions.Reaction>
+                          ))}
+                      </MessageReactions>
+                    </>
+                  ) : (
+                    <>
+                      {msg.t === 'ul' && (
+                        <Message.Header>
+                          <Message.Name>@{msg.u.username} </Message.Name>
+                          <Message.Username style={{ marginLeft: '2px' }}>
```

Comment:
You can see that all these blocks contain exactly the same code except the username part, so IMO we can just make a function of a sort where we can have a switch case to check what is the condition and according to that return the `<Message.Username>{userActions(msg.t)}</Message.Username>`

Something like,
```
function userActions(type) { // its msg.t, please assign it a better name if you think of any
  switch type {
    case 'uj':
        return 'joined the channel' // something like this
    ...
  }
}
```

What do you think?

Reply:
Yes, this would make more sense. Thanks!